### ISSUE TITLE
Re-calculate color contrast of text with shadow

### ISSUE THREAD

**Issue by shunguoy** [IBMa/equal-access#1404]
currently, the engine throws a potential violation when (1) the color contrast of the text (without considering the shadow effects) doesn't satisfy the minimum requirement; and (2) the text has shadow.  A more accurate contrast calculation is needed.

test case:[ ACT Passed Example 4](https://www.w3.org/WAI/standards-guidelines/act/rules/afw4f7/proposed/)

**Comment by philljenkins** [comment]
The [ACT Passed Example 4](https://www.w3.org/WAI/standards-guidelines/act/rules/afw4f7/proposed/) test case is very hard to read, so there may be an issue with the interpretation of the spec or ACT's calculation .
